### PR TITLE
test: Delete site E2E case supports native dialogs

### PR DESCRIPTION
**Pull request by dcalhoun**
Fixes https://github.com/Automattic/dotcom-forge/issues/7236.

## Proposed Changes

- Mock the native delete dialog interaction, as Playwright does not currently
  support interactions with this OS UI.
- Refactor onboarding interactions to improve test stability.

## Testing Instructions

N/A, no user-facing changes.

## Pre-merge Checklist

- [x] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: e2e/page-objects/onboarding.ts [hunk 1]
```diff
@@ -5,11 +5,7 @@ export default class Onboarding {
 	constructor( private page: Page ) {}
 
 	private get locator() {
-		// This fails, not sure why...
-		// return this.page.getByTestId( 'onboarding' );
```
Comment (dcalhoun) [inline e2e/page-objects/onboarding.ts:9]: My guess is that the unexpected failure may have related to attempting queries before the UI was visible. With the new approach proposed in these changes, the original query appears to work consistently.

File: e2e/page-objects/onboarding.ts [hunk 2]
```diff
@@ -29,7 +25,7 @@ export default class Onboarding {
 	}
 
 	get continueButton() {
-		return this.locator.getByRole( 'button', { name: 'Continue' } );
+		return this.locator.getByRole( 'button', { name: 'Add site' } );
```
Comment (dcalhoun) [inline e2e/page-objects/onboarding.ts:28]: Updated to match changes from https://github.com/Automattic/studio/pull/111.

File: e2e/sites.test.ts [hunk 3]
```diff
@@ -24,30 +24,25 @@ test.describe( 'Servers', () => {
...
-		// However, here we are using visibility to trigger onboarding, so we can't use toBeVisible(), which would fail the test.
-		await onboarding.heading.isVisible();
-		if ( await onboarding.heading.isVisible() ) {
-			await expect( onboarding.siteNameInput ).toHaveValue( defaultOnboardingSiteName );
-			await expect( onboarding.localPathInput ).toBeVisible();
-			await expect( onboarding.continueButton ).toBeVisible();
+		const sidebar = new MainSidebar( onboardingMainWindow );
+
+		// Await UI visibility before proceeding.
+		await expect( onboarding.heading.or( sidebar.addSiteButton ) ).toBeVisible();
```
Comment (dcalhoun) [inline e2e/sites.test.ts:41]: By relying upon `toBeVisible`'s interval-based query and the `or` query, this approach should guarantee the app UI is visible before proceeding with interactions.

**Review by derekblank — APPROVED** [review]
LGTM. I observed the e2e tests have passed (both locally and on CI), and the code changes make sense. 🚀 

Screenshot 2024-05-30 at 3 04 53 pm

**Comment by wojtekn** [comment]
@dcalhoun changes look good and work fine.

It caught me at first as I couldn't make e2e work locally today, either in this branch or trunk. I noticed that Studio was using x64 binary, and then I realized I needed to clean `out/` and build fresh binary first. I opened another PR https://github.com/Automattic/studio/pull/190 to improve readme.

**Review by wojtekn — APPROVED** [review]

**Pull request merged**